src/Component/Order/Processor/SplitOrderByVendorProcessor.php

```diff
@@ -34,8 +34,14 @@ public function __construct(
     public function process(OrderInterface $order): array
     {
         $isPrimaryOrder = $order->isPrimary() && 0 < $order->getSecondaryOrders()->count();
+
         if ($isPrimaryOrder) {
-            return [$order, ...$order->getSecondaryOrders()];
+            $orders = [$order, ...$order->getSecondaryOrders()];
+            foreach ($orders as $iOrder) {
```

Comment:
`$iOrder` us not a good name. I would recommend you:

```suggestion
            $secondaryOrders = [$order, ...$order->getSecondaryOrders()];
            foreach ($secondaryOrders as $secondaryOrder) {
```